seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/SinkFlowLifeCycle.java

```diff
@@ -284,18 +287,11 @@ public void restoreState(List<ActionSubtaskState> actionStateList) throws Except
                             .collect(Collectors.toList());
         }
         if (states.isEmpty()) {
-            this.writer =
-                    sinkAction
-                            .getSink()
-                            .createWriter(
-                                    new SinkWriterContext(indexID, metricsContext, eventListener));
+            this.writerContext = new SinkWriterContext(indexID, metricsContext, eventListener);
```

Comment:
we can move it into outer of if statement.

Reply:
You're right, it was my negligence